### PR TITLE
Rewritten the Working With Sessions part of the tutorial

### PR DESCRIPTION
**Pull request by alixdamman**
What bother me with sessions currently is that we don't have auto-completion on items in Pycharm. 
But that's another discussion...

I also figured out how to get links between .rst files and notebooks (thanks to the update of the documentation of nbsphinx). Well, at least, it works on my machine.

Readthedocs output : https://larray-test.readthedocs.io/en/tutorial/tutorial/tutorial_sessions.html

### CODE REVIEW

File: doc/source/tutorial/tutorial_presenting_larray_objects.ipyml [hunk 1]
```diff
@@ -306,39 +296,63 @@ cells:
     A ``Session`` object is a dictionary-like object used to gather several arrays, axes and groups. 
     A session is particularly adapted to gather all input objects of a model or to gather the output arrays from different scenarios. Like with arrays, it is possible to associate metadata to sessions.
     
-    In addition to act like dictionaries, sessions offer several specific methods, like:
-    - ``save`` and ``load`` to save and load all arrays of a session at once in/from CSV or Excel or HDF5 file(s),
-    - ``equals`` and ``array_equals`` to compare arrays between two sessions (scenarios) one by one,
-    - ``apply`` to apply a function to all arrays of a session.
+    See the [Session](../api.rst#session) section of the API Reference to explore all methods of Session objects.
```
Comment (gdementen) [inline doc/source/tutorial/tutorial_presenting_larray_objects.ipyml]: Links to API reference should be **after** presenting each kind of object. and the other links (to load/dump) and general working should come after this section too I think.

File: doc/source/tutorial/tutorial_presenting_larray_objects.ipyml [hunk 2]
```diff
@@ -90,12 +89,19 @@ cells:
...
 
 - markdown: |
     ## LArray
     
-    A ``LArray`` object represents a multidimensional array with labeled axes (#ref needed).
+    A ``LArray`` object represents a multidimensional array with labeled axes.
+    
+    See the [LArray](../api.rst#larray) section of the API Reference to explore all methods of LArray objects.
+    
+    To know how to save and load arrays in CSV, Excel or HDF format, please refer to the [Loading and Dumping Arrays](tutorial_IO.ipynb#Loading-and-Dumping-Arrays) section of the tutorial.
```
Comment (gdementen) [inline doc/source/tutorial/tutorial_presenting_larray_objects.ipyml]: move those links after the explanation.

File: doc/source/tutorial/tutorial_presenting_larray_objects.ipyml [hunk 3]
```diff
@@ -306,39 +296,63 @@ cells:
...
+    # add arrays to the session
+    s_pop.pop = zeros((gender, time))
+    s_pop.births = zeros((gender, time))
+    s_pop.deaths = zeros((gender, time))
     
     # add metadata after creation
-    s.meta.title = 'Input objects'
-    s.meta.description = 'Input axes and arrays for the model X'
+    s_pop.meta.title = 'Demographic Model of Belgium'
+    s_pop.meta.description = 'Modelize the demography of Belgium'
```
Comment (gdementen) [inline doc/source/tutorial/tutorial_presenting_larray_objects.ipyml]: s/Modelize/Models/  Modelize is a word which exists but is not really used.

File: doc/source/tutorial/tutorial_sessions.ipyml [hunk 4]
```diff
@@ -30,95 +28,273 @@ cells:
...
+    compare(s_pop, s_pop_alternative, names=['baseline', 'lower_birth_rate'])
     ```
+    
+    ![compare two sessions](../_static/compare_tutorial.png)
+
+
+- markdown: |
+    <div class="alert alert-warning">
+    
+    **Warning:** Session must have the same list of arrays.
```
[the remark this answers is not in the document]
Comment (alixdamman) [reply doc/source/tutorial/tutorial_sessions.ipyml]: Good to know. I will remove the warning.

File: doc/source/tutorial/tutorial_sessions.ipyml [hunk 5]
```diff
@@ -30,95 +28,273 @@ cells:
...
+    multiplicator[s_pop.odd_years] = 0
+    multiplicator
 
-  id: 7
 
 - code: |
-    s1.element_equals(s3)
+    # multiply all arrays of a session by a common array
+    # here, we set values corresponding to odd years to 0
+    s_pop_even_years = s_pop * multiplicator
```
[the remark this answers is not in the document]
Comment (alixdamman) [reply doc/source/tutorial/tutorial_sessions.ipyml]: We have only one example dataset ('Demography'). In future, I hope we will be able to add new example datasets like 'Pension' for example. In the meantime, I'll try to find another example.

File: doc/source/tutorial/tutorial_sessions.ipyml [hunk 6]
```diff
@@ -30,95 +28,273 @@ cells:
...
+    with another session:
 
-  id: 8
 
 - code: |
-    s1.element_equals(s4)
+    # create a session representing the results of the same model but with 
+    # a lower birth rate in 2015
+    s_pop_alternative = Session(filepath_hdf)
+    s_pop_alternative.births[2015] = (s_pop_alternative.births[2015] * 0.95).astype(int)
```
[the remark this answers is not in the document]
Comment (alixdamman) [reply doc/source/tutorial/tutorial_sessions.ipyml]: Variables here have no axis age, so I don't know how you expect to modify the array pop in a consistent way? In this case, I considered that the births of 2015 are not included in the population of 2015.

File: doc/source/tutorial/tutorial_sessions.ipyml [hunk 7]
```diff
@@ -147,44 +141,36 @@ cells:
...
 - code: |
-    # define a function to add extra years
-    def add_extra_years(array, nb_years):
-        last_year = array.time.i[-1]
-        last_value = array[last_year]
-        # add extra years to the time axis
-        time = array.time.extend(range(last_year+1, last_year+nb_years+1))
-        # duplicate the value from the last year
-        return array.reindex('time', time, fill_value=last_value)
+    # define a function to change the dtype of an array
```
Comment (gdementen) [inline doc/source/tutorial/tutorial_sessions.ipyml]: FWIW, one need not define a new function for that. we can simply use .apply(LArray.astype, dtype=int). Defining a new explicit function is a good idea for the sake of the example but either it should be a different example (a function which does not already exists) or a comment should be present to say that we define the function for the sake of the example.

**Pull request merged**